src/Configuration/Programmer/Programmer_helper.cpp

```diff
@@ -223,6 +259,22 @@ std::vector<Device> extractDeviceList(const std::string& devicesString) {
                  ' ');  // idCode string in hex
     device.tapInfo.idCode = ConvertIntegerStringToInt64(idCodeStr, 0, 16);
 
+    std::string irLenStr;
+    std::getline(lineStream >> std::ws, irLenStr, ' ');  // irlength string
+    device.tapInfo.irLen = ConvertIntegerStringToInt(irLenStr);
+    std::getline(lineStream, firstToken, ' ');  // "found string"
+
+    std::string indexStr;
+    std::getline(lineStream, indexStr, ' ');  // index string
+    device.index = ConvertIntegerStringToInt(indexStr);
+
+    std::getline(lineStream >> std::ws, device.name, ' ');  // device name
+
+    std::string idCodeStr;
+    std::getline(lineStream >> std::ws, idCodeStr,
+                 ' ');  // idCode string in hex
+    device.tapInfo.idCode = ConvertIntegerStringToInt64(idCodeStr, 0, 16);
```

Comment:
so the idcode is int64? it should be u32.

Reply:
the idCode is unsigned32. Probably a static cast is needed.